### PR TITLE
Get actual Homebrew directory with `brew --prefix`

### LINKED ISSUE #1712: Wrong library paths on macOS with Apple Silicon chip
Problem description
------------
On new architectures (Apple Silicon) Homebrew uses `/opt/homebrew/` as default installation directory instead of `/usr/local/`, which it uses on macOS Intel. See https://docs.brew.sh/Installation.

As a consequence, our library paths are wrong on Apple Silicon. This breaks the OpenMP support with the error
```python
epyccel/test_epyccel_openmp.py F

========================================================= FAILURES =========================================================
________________________________________________ test_directive_in_else[c] _________________________________________________

language = 'c'

    @pytest.mark.external
    def test_directive_in_else(language):
>       f1 = epyccel(openmp.directive_in_else, fflags = '-Wall', accelerators=['openmp'], language=language)

epyccel/test_epyccel_openmp.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../pyccel/epyccel.py:376: in epyccel
    mod, fun = epyccel_seq( python_function_or_module, **kwargs )
../pyccel/epyccel.py:225: in epyccel_seq
    execute_pyccel(pymod_filename,
../pyccel/codegen/pipeline.py:372: in execute_pyccel
    generated_filepath = create_shared_library(codegen,
../pyccel/codegen/python_wrapper.py:197: in create_shared_library
    sharedlib_filepath = src_compiler.compile_shared_library(wrapper_compile_obj,
../pyccel/codegen/compiling/compilers.py:458: in compile_shared_library
    self.run_command(cmd, verbose)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cmd = ['/usr/bin/gcc', '-shared', '-O3', '-funroll-loops', '-fPIC', '-Xpreprocessor', ...], verbose = False

    @staticmethod
    def run_command(cmd, verbose):
        """
        Run the provided command and collect the output.
    
        Run the provided compilation command, collect the output and raise any
        necessary errors if the file does not compile.
    
        Parameters
        ----------
        cmd : list of str
            The command to run.
        verbose : bool
            Indicates whether additional output should be shown.
    
        Returns
        -------
        str
            The exact command that was run.
    
        Raises
        ------
        RuntimeError
            Raises `RuntimeError` if the file does not compile.
        """
        cmd = [os.path.expandvars(c) for c in cmd]
        if verbose:
            print(' '.join(cmd))
    
        with subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
                universal_newlines=True) as p:
            out, err = p.communicate()
    
        if verbose and out:
            print(out)
        if p.returncode != 0:
            err_msg = "Failed to build module"
            err_msg += "\n" + err
>           raise RuntimeError(err_msg)
E           RuntimeError: Failed to build module
E           clang: error: no such file or directory: '/usr/local/opt/libomp/lib'

../pyccel/codegen/compiling/compilers.py:500: RuntimeError
--------------------------------------------------- Captured stdout call ---------------------------------------------------

ERROR at shared library generation stage
------------------------------------------------- Captured stdout teardown -------------------------------------------------
Tearing down!
```

Diagnosis
--------
The following parameters are hardcoded in our compiler information, regardless of the actual installation directory of Homebrew:
```
"openmp": {
        "libdirs": [
            "/usr/local/opt/libomp/lib"
        ],
        "includes": [
            "/usr/local/opt/libomp/include"
        ]
    },
```

Suggested fix
--------
Get the correct paths from the following environment variables:
```shell
HOMEBREW_PREFIX
HOMEBREW_CELLAR
HOMEBREW_REPOSITORY
```

### PR DESCRIPTION
**Pull request by yguclu**
Do not hardcode Homebrew installation directory when using Pyccel on macOS platforms. Instead, use command `brew --prefix`. This fixes #1712.

### CODE REVIEW

**Comment by yguclu** [comment]
Most points in the bot checklist do not really apply to this PR. I have checked that the OpenMP tests now work on Apple Silicon chips (on `devel` they fail because of the wrong path).

File: pyccel/compilers/default_compilers.py [hunk 1]
```diff
@@ -110,12 +116,19 @@
                 },
             'family': 'GNU',
             }
+
 if sys.platform == "darwin":
-    gcc_info['openmp']['flags'] = ("-Xpreprocessor",'-fopenmp')
-    gcc_info['openmp']['libs'] = ('omp',)
-    gcc_info['openmp']['libdirs'] = ('/usr/local/opt/libomp/lib',)
-    gcc_info['openmp']['includes'] = ('/usr/local/opt/libomp/include',)
+    p = subprocess.run([shutil.which('brew'), '--prefix'], check=True, capture_output=True)
```
[the remark this answers is not in the document]
Comment (yguclu) [reply pyccel/compilers/default_compilers.py:121]: /bot accept This code is used on macOS.

**Review by EmilyBourne — CHANGES_REQUESTED** [review]
I think this is worth mentioning in the CHANGELOG

**Comment by yguclu** [comment]
> I think this is worth mentioning in the CHANGELOG

Done in 33ea297. Is that sufficient for you?

**Comment by EmilyBourne** [comment]
> > I think this is worth mentioning in the CHANGELOG
> 
> Done in [33ea297](https://github.com/pyccel/pyccel/commit/33ea297fce85cdc53b671f2b5abd25180be93505). Is that sufficient for you?

It is ok but I think it would be better to reference the issue. If this is a problem with OpenMP then I think something like:
```
-   #1712 : Fix library path and OpenMP support for recent Apple chips by getting Homebrew directory with `brew --prefix`.
```

But this may be a little long. I imagine if someone is looking for whether/when OpenMP is fixed they would search OpenMP not Homebrew. The cause of this bug is not necessarily obvious to the user in my opinion so I am not sure that they would be able to connect it to this message.

**Review by EmilyBourne — APPROVED** [review]

**Pull request merged**